### PR TITLE
Threshold calculation adjust

### PR DESCRIPTION
**Pull request by danielSanchezQ**

### CODE REVIEW

**Review by Zeegomo — COMMENTED** [review]
Failure in test is because this PR is coming from a fork

File: scripts/python/proposers_rewards.py [hunk 1]
```diff
@@ -251,23 +259,34 @@ def extract_yes_no_votes(proposal: Proposal, voteplan_proposal: ProposalStatus):
...
 
 def calc_approval_threshold(
-    proposal: Proposal, voteplan_proposal: ProposalStatus, threshold: float
+    proposal: Proposal,
+    voteplan_proposal: ProposalStatus,
+    threshold: float,
+    total_stake_threshold: float,
 ) -> Tuple[int, bool]:
     yes_result, no_result = extract_yes_no_votes(proposal, voteplan_proposal)
+    total_stake = yes_result + no_result
```
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:268]: Uhm, you are completely right we have. But iirc you can not actually vote for it. Also from requirements we do absolutely nothing about it.

File: scripts/python/proposers_rewards.py [hunk 2]
```diff
@@ -251,23 +259,34 @@ def extract_yes_no_votes(proposal: Proposal, voteplan_proposal: ProposalStatus):
...
+    pass_total_threshold = total_stake >= total_stake_threshold
     diff = yes_result - no_result
-    success = diff >= (no_result * threshold)
+    normalized_stake = diff / total_stake_threshold
+    pass_relative_threshold = normalized_stake >= threshold
```
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:272]: Relative one just checks the proportion of yes votes over the total and that should be bigger than the threshold. 

> I find it a bit hard to read (wouldn't pass_relative_threshold = yes >= (1 + threshold) * no be easier?)

Actually, I find yours more difficult 😅

>  but is also probable that I've misunderstood this / requirements have changed.

Yeah they did. You can check jira ticket (607)

File: scripts/python/proposers_rewards.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Zeegomo) [reply scripts/python/proposers_rewards.py:268]: The requirements mention that at least `X` amount of stake should vote on a proposal. I think we should count blank votes here as well (I guess it's easy to do), the limitation is only from the app, third party tools could in theory vote for that option

[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:268]: Sure, it doesn't hurt to have, so I'll do it.

File: scripts/python/proposers_rewards.py [hunk 3]
```diff
@@ -242,32 +250,46 @@ def sanity_check_data(
...
 ) -> Tuple[int, bool]:
-    yes_result, no_result = extract_yes_no_votes(proposal, voteplan_proposal)
+    blank_result, yes_result, no_result = extract_yes_no_votes(
+        proposal, voteplan_proposal
+    )
+    total_stake = blank_result + yes_result + no_result
+    pass_total_threshold = total_stake >= total_stake_threshold
     diff = yes_result - no_result
-    success = diff >= (no_result * threshold)
+    pass_relative_threshold = diff >= (no_result * threshold)
```
Comment (Zeegomo) [inline scripts/python/proposers_rewards.py:275]: I think according to the specification it should be `diff > threshold` (the difference between 'yes' and 'no' percentages should be more than the threshold)

**Review by Zeegomo — CHANGES_REQUESTED** [review]

File: scripts/python/proposers_rewards.py [hunk 4]
```diff
@@ -409,6 +439,8 @@ class OutputFormat(enum.Enum):
 def calculate_rewards(
     conversion_factor: float = typer.Option(...),
     output_file: str = typer.Option(...),
+    block0_path: str = typer.Option(...),
+    total_stake_threshold: float = typer.Option(0.1),
```
[the remark this answers is not in the document]
Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:443]: good catch!

File: scripts/python/proposers_rewards.py (hunk 3 above)
Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:275]: Lest ask this, because I remember it to be "at least" the threshold.

Comment (Zeegomo) [reply scripts/python/proposers_rewards.py:275]: Yes, my mistake. But besides  `> / >=` I actually meant that the calculation is different. The threshold is not on the ratio between 'yes' and 'no' but on the absolute difference between the two

Comment (danielSanchezQ) [reply scripts/python/proposers_rewards.py:275]: Talked with @MarklarIO and it is:
```
IF(B3/C3>=1.15,"YES","NO")

where
B3 is YES votes total per proposal
C3 is NO votes total per proposal
```

**Pull request merged**